sale_order_revision/models/sale_order.py

```diff
@@ -121,3 +123,19 @@ def create_revision(self):
 
         # Returning the new sale order view with new record.
         return action
+
+    def action_confirm(self):
+        for order in self:
+            current_revision = order.current_revision_id
+            if current_revision:
+                raise UserError(
+                    _(
+                        "Order {order} has a new Revision {revision}:\n"
```

Comment:
In fact, I come back on this. This is better:

```suggestion
                        "Order %(order)s has a new Revision %(revision)s:\n"
```